dkist/dataset/loader.py

```diff
@@ -14,6 +15,12 @@
     from asdf import ValidationError
 
 
+def asdf_open_memory_mapping_kwarg(memmap: bool) -> dict:
+    if tuple(int(part) for part in importlib.metadata.version("asdf").split(".")) >= (3, 1, 0):
```

Comment:
@zacharyburnett it looks like the version check is failing with a dev version:
https://github.com/DKISTDC/dkist/actions/runs/10288624099/job/28474506653?pr=422#step:10:1080
Maybe a string comparison is "good enough":
```suggestion
    if asdf.__version__ > "3.1.0":
```

This does have the issue that "3.1.0.dev" is greater than "3.1.0" but since we're past that version I think that's fine.